### PR TITLE
Add guide about custom API authentication

### LINKED ISSUE #8123: Guide about  API authentication (including custom)
**Description**

We should write a guide that describes how to add new ones or use the current authentication method.

More information about the docs for REST API is available:
[Docs for REST API](https://github.com/apache/airflow/issues/8143)

**Use case / motivation**

N/A

**Related Issues**

N/A

### PR DESCRIPTION
**Pull request by mik-laj**
Close: https://github.com/apache/airflow/issues/8123

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by mik-laj** [comment]
CC: @jhtimmins

File: docs/security.rst [hunk 1]
```diff
@@ -160,6 +160,24 @@ look like the following.
           -H 'Cache-Control: no-cache' \
           -H "Authorization: Bearer ${ID_TOKEN}"
 
+Roll your own API authentication
+''''''''''''''''''''''''''''''''
+
+Each auth backend is defined as a new Python module. It must have 2 defined methods:
```
Comment (ephraimbuddy) [inline docs/security.rst:166]:
```suggestion
Each auth backend should be defined as a new Python module. It must have 2 defined methods:
```

File: docs/security.rst [hunk 2]
```diff
@@ -160,6 +160,24 @@ look like the following.
...
+
+* ``init_app(app: Flask)`` - function invoked when creating a flash application, which allows you to add a new view.
```
Comment (ephraimbuddy) [inline docs/security.rst]:
```suggestion
* ``init_app(app: Flask)`` - function invoked when creating a flask application, which allows you to add a new view.
```

**Review by kaxil — APPROVED** [review]

File: docs/security.rst (hunk 2 above)
Comment (kaxil) [reply docs/security.rst]: +1

**Pull request merged**